book/bayesian_optimization.qmd

```diff
@@ -0,0 +1,844 @@
+---
+author:
+  - name: Lennart Schneider
+    orcid: 0000-0003-4152-5308
+    email: lennart.schneider@stat.uni-muenchen.de
+    affiliations:
+      - name: Ludwig-Maximilians-Universität München
+abstract:
+  Many optimization problems are a black box, i.e., the only observable information is the output value of the function given an input value.
+  A prime example is given by hyperparameter optimization where we configure a learner with a hyperparameter configuration and estimate its generalization performance via resampling.
+  Moreover, hyperparameter optimization can be extremely expensive and therefore calls for very sample efficient optimization algorithms.
+  This chapter presents an introduction to Bayesian optimization (also called model-based optimization) within the mlr3 ecosystem.
+  We start with the general black box optimization setting and proceed to introduce the building blocks of Bayesian optimization - a class of sample efficient iterative global optimization algorithms.
+  After that we show that Bayesian optimization can be used out of the box for hyperparameter optimization of machine learning algorithms.
+  We further show how multi-objective Bayesian optimization can be performed and briefly discuss noisy objective functions.
+  The chapter closes with comments on robustness, safety nets, defaults and practical considerations regarding Bayesian optimization algorithms.
+---
+
+# Bayesian Optimization {#sec-bayesian-optimization}
+
+```{r}
+set.seed(1005)
+```
+
+{{< include _setup.qmd >}}
+
+Black box optimization considers the optimization of a so-called black box function whose structure and analytical description is unknown, unexploitable or non-existent.
+As a result, the only observable information is the output value of the function given an input value.
+Usually, the objective function is subject so some box constraint, i.e., in the case of numeric variables the domain is assumed to be a hyperrectangle.
+Examples for black box optimization problems range from real-life experiments, e.g., crash tests or chemical reactions to computer simulations.
```

Comment:
Some of this is already covered in C4, as you note below, so would instead linking back to C4 in the first line and then generalising this to black-box more generally